servers/src/main/java/tachyon/worker/netty/protocol/buffer/DataBuffer.java

```diff
@@ -33,4 +35,11 @@
    * @return the length of the data in bytes.
    */
   public abstract long getLength();
+
+  /**
+   * Returns a {@link ByteBuffer} for read-only access to the data.
+   *
+   * @return a read-only ByteBuffer representing the data
+   */
+  public abstract ByteBuffer getReadOnlyByteBuffer();
```

Comment:
Not necessarily. This interface could be used for the client side as well, when that is implemented. I'm not committed to the exactly API, so, it may change slightly when the client is refactored to use netty.

In general, I think there needs to be two ways to read the data: one for Netty output, and one to actually read the bytes.